python/lsst/ctrl/bps/wms/panda/idds_tasks.py

```diff
@@ -0,0 +1,126 @@
+        self.bps_config = config
+        self.tasks_inputs = {}
+        self.himem_tasks = set(config.get("himem_steps"))
+        self.computing_queue = config.get("computing_queue")
+        self.computing_queue_himem = config.get("computing_queue_himem")
+
+        _, v = config.search("maxwalltime", opt={"default": 90000})
+        self.maxwalltime = v
+        _, v = config.search("maxattempt", opt={"default": 5})
+        self.maxattempt = v
+
+    def define_task_name(self, step):
+        return self.bps_config.get('workflowName') + '_' + step
+
+    def pick_non_init_cmdline(self):
+        for node_name in self.bps_workflow.nodes:
+            if node_name != 'pipetaskInit':
+                return self.bps_workflow.nodes[node_name]['job'].cmdline
+
+    def define_tasks(self):
+        tasks = []
+        raw_dependency_map = self.create_raw_jobs_dependency_map()
+        tasks_dependency_map = self.split_map_over_tasks(raw_dependency_map)
+        init_task_cmd_line = self.bps_workflow.nodes['pipetaskInit']['job'].cmdline
+        other_task_cmd_line = self.pick_non_init_cmdline()
+
+        for task_step, jobs in tasks_dependency_map.items():
+            task = LSSTTask()
+            task.step = task_step
+            task.name = self.define_task_name(task.step)
+            task.queue = self.computing_queue_himem if task_step in self.himem_tasks else self.computing_queue
+            task.lfns = list(jobs.keys())
+            task.local_pfns = self.tasks_inputs[task.name]
+            task.maxattempt = self.maxattempt
+            task.maxwalltime = self.maxwalltime
+
+            # We take the commandline only from first job  because PanDA uses late binding and
+            # command line for each job in task is equal to each other in exception to the processing
+            # file name which is substitutes by PanDA
+            if task_step == 'pipetaskInit':
+                task.executable = init_task_cmd_line
+            else:
+                task.executable = other_task_cmd_line
+            tasks.append(task)
+        self.add_dependencies(tasks, tasks_dependency_map)
+        return tasks
+
+    def add_dependencies(self, tasks, tasks_dependency_map):
+        for task in tasks:
+            jobs = tasks_dependency_map[task.step]
+            dependencies = []
+            for job, job_dependency in jobs.items():
+                job_dep = {
+                    "name": job,
+                    "submitted": False,
+                    "dependencies": [{"task": dependency[0], "inputname":
+                                     dependency[1][0], "available": False} for dependency in
+                                     job_dependency.items()]
+                }
+                dependencies.append(job_dep)
+            task.dependencies = dependencies
+
+    def define_execution_command(self):
+        exec_str = ""
+        if self.bps_config.get("computing_queue") == 'docker':
+            pass

```

Comment:
What's the code meant to be doing? Should the `if` be deleted?

Reply:
Code cleaned.